zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterManagedProcess.java

```diff
@@ -99,24 +100,24 @@ public void start(String userName, Boolean isUserImpersonate) {
     // start server process
     final String callbackHost;
     final int callbackPort;
+    TServerSocket tSocket = null;
     try {
-      port = RemoteInterpreterUtils.findRandomAvailablePortOnAllLocalInterfaces(portRange);
-      logger.info("Choose port {} for RemoteInterpreterProcess", port);
+      tSocket = RemoteInterpreterUtils.createTServerSocket(callbackPortRange);
+      callbackPort = tSocket.getServerSocket().getLocalPort();
       callbackHost = RemoteInterpreterUtils.findAvailableHostAddress();
-      callbackPort = RemoteInterpreterUtils.findRandomAvailablePortOnAllLocalInterfaces();
     } catch (IOException e1) {
```

Comment:
catch TTransportException?

Reply:
TTransportException is wrapped in IOException, so there it is enough to catch IOException

Reply:
not from what I can see, which is why I asked:
https://github.com/apache/thrift/blob/master/lib/javame/src/org/apache/thrift/transport/TTransportException.java

Reply:
Sorry, I mean the code in try catch block has wrapped TTransportException in IOException. 
```
tSocket = RemoteInterpreterUtils.createTServerSocket(callbackPortRange);
callbackPort = tSocket.getServerSocket().getLocalPort();
callbackHost = RemoteInterpreterUtils.findAvailableHostAddress();
```